components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -84,11 +92,10 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
 
         LOG.debug("Access token validation success.");
 
-        AccessTokenDO tokenDO = OAuth2Util.findAccessToken(token, false);
+        this.tokenDO = OAuth2Util.findAccessToken(token, false);
```

Comment:
Why we need an instance variable. I hope this should be a local variable. Making instance variable could cause issue when handling multiple requests concurrently. isn't it?

Reply:
Some of the attributes in this token object will be used when revoking the authz_code token. The token revocation part happens in the `issue()` method, so we need to store this variable so that it will be available when calling `issue()` method

Reply:
We can't do that right. I mean kept that in a instance varibale

Reply:
Instance variables are meant to keep the state of the object of the given class. But I feel we are violating something and could cause issues also.